lib/screenplay/pa_messages/pa_message.ex

```diff
@@ -39,4 +40,32 @@ defmodule Screenplay.PaMessages.PaMessage do
 
     timestamps(type: :utc_datetime)
   end
+
+  def changeset(message, attrs \\ %{}) do
+    message
+    |> cast(attrs, [
+      :alert_id,
+      :start_time,
+      :end_time,
+      :days_of_week,
+      :sign_ids,
+      :priority,
+      :interval_in_minutes,
+      :visual_text,
+      :audio_text,
+      :paused,
+      :saved
+    ])
+    |> validate_required([
+      :start_time,
+      :days_of_week,
+      :sign_ids,
+      :priority,
+      :interval_in_minutes,
+      :visual_text,
+      :audio_text
+    ])
+    |> validate_length(:sign_ids, min: 1)
+    |> validate_subset(:days_of_week, 1..7)
+  end
```

Comment:
**question:** should we validate that `end_time` is `nil` if `alert_id` is not `nil`?

Reply:
Good call. Will add.